inginious/frontend/plugins/register_students/pages/api/add_course_students_csv_file_api.py

```diff
@@ -0,0 +1,149 @@
+        else:
+            passwd_hash = hashlib.sha512(data["passwd"].encode("utf-8")).hexdigest()
+            activate_hash = hashlib.sha512(str(random.getrandbits(256)).encode("utf-8")).hexdigest()
+            self.database.users.insert({"username": data["username"],
+                                        "realname": data["realname"],
+                                        "email": data["email"],
+                                        "password": passwd_hash,
+                                        "activate": activate_hash,
+                                        "bindings": {},
+                                        "language": self.user_manager._session.get("language", "en")})
+            try:
+                web.sendmail(web.config.smtp_sendername, data["email"], "Welcome on UNCode",
+                             """Welcome on UNCode!
+                             
+                             To activate your account, please click on the following link:"""
+                             + web.ctx.home + "/register?activate=" + activate_hash
+                             + """\nRemember to change your password in preference section.
+                             
+                             Note: Your password is the same as your username.""")
+            except:
+                pass
+
+        return success
+
+    def _check_email_format(self, email):
+        """Checks email matches a real email."""
+        email_re = re.compile(
+            r"(^[-!#$%&'*+/=?^_`{}|~0-9A-Z]+(\.[-!#$%&'*+/=?^_`{}|~0-9A-Z]+)*"  # dot-atom
+            r'|^"([\001-\010\013\014\016-\037!#-\[\]-\177]|\\[\001-011\013\014\016-\177])*"'  # quoted-string
+            r')@(?:[A-Z0-9](?:[A-Z0-9-]{0,61}[A-Z0-9])?\.)+[A-Z]{2,6}\.?$', re.IGNORECASE)  # domain
+
+        return email_re.match(email)
+
+    def _parse_user_data(self, user_data):
+        """
+        Parses the user data into a dict.
+        :param user_data: array containing the user data, That array comes from the parsed file.
+        :return: Dict containing all parsed information.
+        """
+        name = user_data[0]
+        lastname = user_data[1]
+        email = user_data[2]
+        username = email.split("@")[0]
+
+        data = {
+            "realname": name + " " + lastname,
+            "username": username,
+            "email": email,
+            "passwd": username
+        }
+
+        return data
+
+    def _file_well_formatted(self, parsed_file):
+        """
+        Checks that the email has the required information, with three columns, emails at last column and with the right
+        format.
+        :return: True if the file correctly formatted. Otherwise returns False.
+        """
+        for data in parsed_file:
+            if len(data) != 3:
+                return False
+            elif self._check_email_format(data[2]) is None:
+                return False
+
+        return True
+
+    def _parse_csv_file(self, csv_file):
```

Comment:
This code is ok. But have you check if this can be done with the [csv module](https://docs.python.org/3/library/csv.html)?